runtime-light/scheduler/scheduler.cpp

```diff
@@ -20,32 +21,35 @@ SimpleCoroutineScheduler &SimpleCoroutineScheduler::get() noexcept {
 }
 
 ScheduleStatus SimpleCoroutineScheduler::scheduleOnNoEvent() noexcept {
-  if (yield_coros.empty()) {
+  if (yield_tokens.empty()) {
     return ScheduleStatus::Skipped;
   }
-  const auto coro{yield_coros.front()};
-  yield_coros.pop_front();
-  coro.resume();
+  const auto token{yield_tokens.front()};
+  yield_tokens.pop_front();
+  suspend_tokens.erase(token);
+  token.first.resume();
```

Comment:
`SimpleCoroutineScheduler` is simple :) It's just an event based scheduler that prioritizes events from the platform. That's why it schedules a coroutine waiting for reschedule and pass control to its caller immediately. In such cases it hopes that there could be new updates from the platform it can schedule on.

Reply:
Also, in case there are coroutines waiting for reschedule and there are not updates from the platform, the scheduler will resume coroutines waiting for reschedule